File: job-offer/src/main/java/com/oxygensend/joboffer/context/user/view/UserView.java
package com.oxygensend.joboffer.context.user.view;


public class UserView extends BaseUserView {
    public final String activeJobPosition;

    public UserView(String id, String fullName, String thumbnailPath, String activeJobPosition) {
        super(id, fullName, thumbnailPath);
        this.activeJobPosition = activeJobPosition;
    }
}

File: opinions/src/main/kotlin/com/oxygensend/opinions/context/view/UserView.kt
package com.oxygensend.opinions.context.view

import com.oxygensend.opinions.context.dto.AuthorDto
import com.oxygensend.opinions.domain.User

data class UserView(
    val id: String,
    val fullName: String?,
    val thumbnailPath: String?
) {
    companion object {
        fun from(author: String) = UserView(
            id = author,
            fullName = null,
            thumbnailPath = null
        )

        fun from(user: User) = UserView(
            id = user.id,
            fullName = user.fullName,
            thumbnailPath = user.thumbnailPath
        )

        fun from(authorDto: AuthorDto) = UserView(
            id = authorDto.id,
            fullName = authorDto.fullName,
            thumbnailPath = authorDto.thumbnailPath
        )
    }
}
